src/languages/haxe.js

```diff
@@ -89,7 +101,7 @@ export default function(hljs) {
       },
       {
         className: 'class', // abstracts
-        beginKeywords: 'abstract',
+        begin: '\\babstract\\b(?=\\s*' + hljs.IDENT_RE + '\\s*\\()',
```

Comment:
Is all this necessary?  Is `abstract` a soft/contextual keyboard in Haxe?

Reply:
Early in Haxe, the `abstract` keyword was only for this specific declaration:
```haxe
abstract SomeName(TYPE) from TYPE to TYPE
```

But now it's a way more common keyword that's used as an special identifier and a descriptor for `class` and `function`. 

Preferably, it should be treated just like an arbitrary keyword at this point,. This code is more meant to setup the correct context for coloring "from" and "to". Those are not reserved keywords and can be used as normal variable names, so should  only be colored in the specific context above.

Maybe there's a better way to do it? Or would it be okay just to have "from" and "to" colored as keywords?

Reply:
This all makes sense given from and to are contextual.